http/obmc_shell.hpp

```diff
@@ -0,0 +1,245 @@
+#pragma once
+
+#include <pty.h>
+#include <pwd.h>
+#include <termios.h>
+
+#include <app.hpp>
+#include <boost/asio/posix/stream_descriptor.hpp>
+#include <boost/process/io.hpp>
+#include <websocket.hpp>
+
+#include <map>
+
+namespace crow
+{
+
+namespace obmc_shell
+{
+
+class Handler : public std::enable_shared_from_this<Handler>
+{
+  public:
+    Handler(crow::websocket::Connection* conn) :
+        session(conn), streamFileDescriptor(conn->getIoContext()),
+        doingWrite(false)
+    {}
+
+    ~Handler() = default;
+
+    void doClose()
+    {
+        streamFileDescriptor.close();
+
+        // boost::process::child::terminate uses SIGKILL, need to send SIGTERM
+        int rc = kill(pid, SIGKILL);
+        session = nullptr;
+        if (rc)
```

Comment:
If kill returns a non-zero value, should we just return? If fail to kill process, then will we leave a dangling process I think. Maybe a try, catch block (but maybe that is too much work for this commit)

Reasons why kill will return non-zero value.

   ERRORS
   EINVAL An invalid signal was specified.

   EPERM  The calling process does not have permission to send the
          signal to any of the target processes.

   ESRCH  The target process or process group does not exist.  Note
          that an existing process might be a zombie, a process that
          has terminated execution, but has not yet been wait(2)ed
          for.

Reply:
EINVAL : when I was sending SIGTERM that time it becomes a zombi process, the only solution to fix this us reboot server. SIGKILL kills the process, so I don't think we are sending the wrong signal.

 EPERM : bmcweb run as the root user so I don't think so this is a problem.
 
 ESRCH : no idea about this but if this happened then yes we might have a zombie process. but I can think what is a solution for that.
 
 try and catch: for system-level call does try-catch work? because they do not throw errors they return errors.